File: SDKs/beezup.com/2.0/typescript/src/sdk/models/operations/optimisebycategory.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import * as shared from "../shared";



export class OptimiseByCategoryPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=actionName" })
  actionName: shared.OptimisationActionNameGeneralParametersEnum;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=catalogCategoryId" })
  catalogCategoryId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=storeId" })
  storeId: string;
}


export class OptimiseByCategoryRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: OptimiseByCategoryPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: string[];
}


export class OptimiseByCategoryResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  beezUPCommonErrorResponseMessage?: shared.BeezUpCommonErrorResponseMessage;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}

File: SDKs/beezup.com/2.0/python/src/sdk/models/operations/optimisebycategory.py
import dataclasses
from typing import Optional
from enum import Enum
from ..shared import optimisationactionnamegeneralparameters_enum as shared_optimisationactionnamegeneralparameters_enum
from ..shared import beezup_common_errorresponsemessage as shared_beezup_common_errorresponsemessage


@dataclasses.dataclass
class OptimiseByCategoryPathParams:
    action_name: shared_optimisationactionnamegeneralparameters_enum.OptimisationActionNameGeneralParametersEnum = dataclasses.field(metadata={'path_param': { 'field_name': 'actionName', 'style': 'simple', 'explode': False }})
    catalog_category_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'catalogCategoryId', 'style': 'simple', 'explode': False }})
    store_id: str = dataclasses.field(metadata={'path_param': { 'field_name': 'storeId', 'style': 'simple', 'explode': False }})
    

@dataclasses.dataclass
class OptimiseByCategoryRequest:
    path_params: OptimiseByCategoryPathParams = dataclasses.field()
    request: Optional[list[str]] = dataclasses.field(default=None, metadata={'request': { 'media_type': 'application/json' }})
    

@dataclasses.dataclass
class OptimiseByCategoryResponse:
    content_type: str = dataclasses.field()
    status_code: int = dataclasses.field()
    beez_up_common_error_response_message: Optional[shared_beezup_common_errorresponsemessage.BeezUpCommonErrorResponseMessage] = dataclasses.field(default=None)
    
